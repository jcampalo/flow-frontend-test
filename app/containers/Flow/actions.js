import { actionTypes as at } from './constants';

export const addRule = () => ({
  type: at.ADD_RULE
});

export const changeRule = ({ name, value, index }) => ({
  type: at.CHANGE_RULE,
  payload: { name, value, index }
});

export const changeContent = ({ value }) => ({
  type: at.CHANGE_CONTENT,
  payload: { value }
});

export const execute = () => ({
  type: at.EXECUTE
});

export const executeEnd = ({ results }) => ({
  type: at.EXECUTE_END,
  payload: { results }
});
