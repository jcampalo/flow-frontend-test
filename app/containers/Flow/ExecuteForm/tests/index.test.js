import React from 'react';
import { create } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import { FlowExecuteForm, mapDispatchToProps } from '../index';
import { changeContent, execute } from '../../actions';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <IntlProvider {...props}>
      <FlowExecuteForm {...finalProps} />
    </IntlProvider>
  );
};

describe('<FlowExecuteForm />', () => {
  it('should render component', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChange', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChange).toBeDefined();
      });

      it('should dispatch changeContent when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const value = 'value';

        result.onChange(value);

        expect(dispatch).toHaveBeenCalledWith(changeContent({ value }));
      });
    });

    describe('onSubmit', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(result.onSubmit).toBeDefined();
      });

      it('should dispatch execute when called', () => {
        const dispatch = jest.fn();
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const evt = { preventDefault };

        result.onSubmit(evt);

        expect(preventDefault).toHaveBeenCalledWith();
        expect(dispatch).toHaveBeenCalledWith(execute());
      });
    });
  });
});
