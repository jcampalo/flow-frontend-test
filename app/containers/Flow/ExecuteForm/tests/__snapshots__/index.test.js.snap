// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlowExecuteForm /> should render component 1`] = `
<form
  className="Form__Form-PASBS gSSxyJ"
  onSubmit={undefined}
>
  <label
    htmlFor="content"
  >
    <span
      className="Wrapper-dyuznp gAhBoi"
    >
      <label
        className="Label-ghyrbC KoSFt"
      >
        Content
      </label>
      <textarea
        autoComplete="off"
        className="StyledTextArea__TextArea-cqmHOB jGwssY"
        name="content"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        rows={6}
        type="text"
        value={undefined}
      />
    </span>
  </label>
  <div
    className="SubmitButton__SubmitButton-iZeMNu fhFgOJ Wrapper__Wrapper-cZxMuO eDyakP"
  >
    <button
      className="StyledButton__StyledButton-iQSVZX hrAcih"
      disabled={true}
      onClick={undefined}
      type="submit"
    >
      <span>
        Run
      </span>
    </button>
  </div>
</form>
`;
